tmail-backend/saas/saas-api/src/main/java/com/linagora/tmail/saas/mailet/I18NDSNBounce.java

```diff
@@ -0,0 +1,174 @@
+    private static final ImmutableSet<String> CONFIGURABLE_PARAMETERS = ImmutableSet.of(
+        "debug", "passThrough", "attachment", "sender", "prefix", "action", "defaultStatus",
+        SUPPORTED_LANGUAGES_PARAMETER, I18N_DSN_TEMPLATE_DIRECTORY_PARAMETER);
+    private static final String DEFAULT_I18N_DSN_TEMPLATE_DIRECTORY = "classpath://eml-template/dsn/";
+    private static final Locale DEFAULT_LANGUAGE = Locale.ENGLISH;
+
+    private final FileSystem fileSystem;
+    private final JmapSettingsRepository jmapSettingsRepository;
+    private Set<Locale> supportedLanguages;
+    private String i18nDsnTemplateDirectory;
+
+    @Inject
+    public I18NDSNBounce(DNSService dns,
+                         FileSystem fileSystem,
+                         JmapSettingsRepository jmapSettingsRepository) {
+        super(dns);
+        this.fileSystem = fileSystem;
+        this.jmapSettingsRepository = jmapSettingsRepository;
+    }
+
+    @Override
+    public void init() throws MessagingException {
+        super.init();
+        supportedLanguages = asSupportedLanguages(getInitParameter(SUPPORTED_LANGUAGES_PARAMETER));
+        i18nDsnTemplateDirectory = getInitParameter(I18N_DSN_TEMPLATE_DIRECTORY_PARAMETER, DEFAULT_I18N_DSN_TEMPLATE_DIRECTORY);
+    }
+
+    @Override
+    public Set<String> getAllowedInitParameters() {
+        return CONFIGURABLE_PARAMETERS;
+    }
+
+    @Override
+    protected MimeBodyPart createTextMsg(Mail originalMail) throws MessagingException {
+        StringBuilder builder = new StringBuilder();
+
+        builder.append(i18nBounceMessage(originalMail)).append(LINE_BREAK);
+
+        Optional.ofNullable(originalMail.getMessage().getSubject())
+            .ifPresent(subject -> builder.append("Original email subject: ")
+                .append(subject)
+                .append(LINE_BREAK)
+                .append(LINE_BREAK));
+        builder.append(action.asString()).append(" recipient(s):").append(LINE_BREAK);
+        builder.append(originalMail.getRecipients()
+            .stream()
+            .map(MailAddress::asString)
+            .collect(Collectors.joining(", ")));
+        builder.append(LINE_BREAK).append(LINE_BREAK);
+        if (action.shouldIncludeDiagnostic()) {
+            Optional<String> deliveryError = AttributeUtils.getValueAndCastFromMail(originalMail, DELIVERY_ERROR, String.class);
+
+            deliveryError.or(() -> Optional.ofNullable(originalMail.getErrorMessage()))
+                .ifPresent(message -> {
+                    builder.append("Error message:").append(LINE_BREAK);
+                    builder.append(message).append(LINE_BREAK);
+                    builder.append(LINE_BREAK);
+                });
+        }
+        MimeBodyPart bodyPart = new MimeBodyPart();
+        bodyPart.setText(builder.toString());
+        return bodyPart;
+    }
+
+    private String i18nBounceMessage(Mail originalMail) {
+        getSenderLocaleFromSettings(originalMail);
```

Comment:
If the sender is not a local user we should lookup the first `Content-Language` header of the inbound mail, recursing through the mime body part structure....